qt/python/mantidqt/widgets/src/_widgetscore.sip

```diff
@@ -0,0 +1,98 @@
+%ModuleCode
+#include "MantidQtWidgets/Common/Message.h"
+#include "MantidAPI/IAlgorithm.h"
+#include "MantidQtWidgets/Common/AlgorithmDialog.h"
+// Allows suppression of namespaces within the module
+using namespace MantidQt::MantidWidgets;
+using namespace MantidQt::API;
+using namespace Mantid::API;
+%End
+
+%InitialisationCode
+qRegisterMetaType<MantidQt::MantidWidgets::Message>();
+%End
+
+// ----------------------------------------------------------------------------
+// Classes
+// ----------------------------------------------------------------------------
+
+class Configurable {
+%TypeHeaderCode
+#include "MantidQtWidgets/Common/Configurable.h"
+%End
+
+public:
+  void readSettings(const QSettings &storage);
+  void writeSettings(QSettings *storage);
+
+private:
+  // Not constructible or copyable
+  Configurable();
+  Configurable(const Configurable&);
+};
+
+class MessageDisplay : QWidget, Configurable {
+%TypeHeaderCode
+#include "MantidQtWidgets/Common/MessageDisplay.h"
+%End
+
+public:
+  MessageDisplay(QWidget *parent = 0);
+  void attachLoggingChannel();
+};
+
+
+class AlgorithmDialog: QDialog {
+%TypeHeaderCode
+#include "MantidQtWidgets/Common/AlgorithmDialog.h"
+#include "MantidAPI/AlgorithmObserver.h"
+#include <boost/python/extract.hpp>
+%End
+public:
+  void setShowKeepOpen(const bool showOption);
```

Comment:
Do we need to expose all of these methods?

Reply:
I don't know.